public/app/features/admin/ldap/LdapPage.tsx

```diff
@@ -163,4 +157,7 @@ const mapDispatchToProps = {
   clearUserMappingInfo,
 };
 
-export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(LdapPage));
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type Props = OwnProps & ConnectedProps<typeof connector>;
```

Comment:
Didn't know the types are hoisted like this, probably some TS magic. Not sure if this can be confusing tho, as it's used before declaration. In other components we define `mapStateToProps` and  `mapDispatchToProps` at the top, but I can see a value in doing it this way as well.